synapse/server_notices/server_notices_manager.py

```diff
@@ -102,6 +102,10 @@ async def maybe_get_notice_room_for_user(self, user_id: str) -> Optional[str]:
         Returns:
             The room's ID, or None if no room could be found.
         """
+        # If there is no server notices MXID, then there is no server notices room
+        if self.server_notices_mxid is None:
+            return None
```

Comment:
Shortcut missing before

Reply:
This method is not called unless `server_notices_mxid` is set (all the callers bail early). I think that's why we didn't add this check?

Reply:
Preference on keeping or removing?

If we expect callers to make the check, perhaps it should just be an assert.

Reply:
No strong opinion! Probably worth keeping since it is a public method? (And I can't imagine an extra if-check would really hurt perf.)